bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/AbstractComponent.java

```diff
@@ -92,14 +94,16 @@ public AbstractComponent(CFactory.ComponentConfiguration componentConfiguration,
                     this.channelConfiguration.payload_not_available);
 
             availablityChannel = buildChannel(HomeAssistantThingHandler.AVAILABILITY_CHANNEL, value,
-                    channelConfiguration.name + " availability").listener(componentConfiguration.getUpdateListener())//
+                    channelConfiguration.name + " availability", componentConfiguration.getUpdateListener())//
```

Comment:
What is the benefit of this change?

Reply:
The update listener used to be Nullable in the channel.
But the channel did not make sense without a listener.
Therefore I moved the listener from the factory method to the constructor and removed the Nullable annotation.
Now we are sure, we have a listener available  when the channel is started.